From 10f5eb3ca40c939537eb9978660f29b39019af40 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Tue, 1 Mar 2022 16:57:22 -0800
Subject: [PATCH] CMake: add dependency on pofiles for default_templates

The GETTEXT_PROCESS_PO_FILES macro from po/CMakeLists.txt creates the
.gmo files from .po files, and it also creates a custom cmake target
named `pofiles`.

Because the target `default_templates` requires that the .gmo files
exist, a dependency on `pofiles` should be declared, thus preventing
possible failures for parallelized builds.

Fixes https://gitlab.com/inkscape/inkscape/-/issues/327
---
 share/templates/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/share/templates/CMakeLists.txt b/share/templates/CMakeLists.txt
index 70665deafa..9eb019a98d 100644
--- a/share/templates/CMakeLists.txt
+++ b/share/templates/CMakeLists.txt
@@ -31,6 +31,9 @@ if(ENABLE_NLS)
   )
   add_custom_target(default_templates ALL DEPENDS default_templates.timestamp)
 
+  # .gmo file creation must occur before creating the templates
+  add_dependencies(default_templates pofiles)
+
   get_inkscape_languages()
   foreach(language_code ${INKSCAPE_LANGUAGE_CODES})
     string(MAKE_C_IDENTIFIER "${language_code}" language_code_escaped)
-- 
GitLab

